articles/user-guide/language/expressions.md

```diff
@@ -567,45 +380,71 @@ let slice9  = arr[...-1...];  // slice9 is [6,5,4,3,2,1];
-concatenation would be expressed as:
+This in particular also includes arrays.
+In particular, it is not possible to update array items. 
+To modify an existing array requires leveraging a *copy-and-update* mechanism.
 
-```qsharp
-(a+b)[13]
-```
-
-All arrays in Q# are zero-based.
-That is, the first element of an array `a` is always `a[0]`.
-
-
-## Copy-and-Update Expressions
-
-New arrays can be created from existing ones via copy-and-update expressions.
-A copy-and-update expression is an expression of the form `expression1 w/ expression2 <- expression3`, where `expression1` has to be of type `T[]` for some type `T`. The second `expression2` defines the indices of the element(s) to modify compared to the array in `expression1` and has to be either of type `Int` or of type `Range`. If `expression2` is of type `Int`, `expression3` has to be of type `T`. If `expression2` is of type `Range`, `expression3` has to be of type `T[]`.
+New arrays can be created from existing ones via *copy-and-update* expressions.
+A copy-and-update expression is an expression of the form `expression1 w/ expression2 <- expression3`, where `expression1` has to be of type `T[]` for some type `T`.
+The second `expression2` defines the indices of the element(s) to modify compared to the array in `expression1` and has to be either of type `Int` or of type `Range`.
+If `expression2` is of type `Int`, `expression3` has to be of type `T`. If `expression2` is of type `Range`, `expression3` has to be of type `T[]`.
 
 A copy-and-update expression `arr w/ idx <- value` constructs a new array with all elements set to the corresponding element in `arr`, except for the element(s) at `idx`, which are set to the one(s) in `value`. 
 For example, if `arr` contains an array `[0,1,2,3]`, then 
 - `arr w/ 0 <- 10` is the array `[10,1,2,3]`.
 - `arr w/ 2 <- 10` is the array `[0,1,10,3]`.
 - `arr w/ 0..2..3 <- [10,12]` is the array `[10,1,12,3]`.
 
+#### Copy-and-update expressions for named items
+
 Similar expressions exist for named items in user-defined types. 
+
 Consider for example the type 
 
 ```qsharp
 newtype Complex = (Re : Double, Im : Double);
 ```
-If `c` contains the value of type `Complex(1.,-1.)`, then `c w/ Re <- 0.` is an expression of type `Complex` that evaluates to `Complex(0.,-1.)`.
+If `c` contains the value of type `Complex(1., -1.)`, then `c w/ Re <- 0.` is an expression of type `Complex` that evaluates to `Complex(0., -1.)`.
+
+### Jagged Arrays
+
+A jagged array, sometimes called an "array of arrays," is an array whose elements are arrays.
+The elements of a jagged array can be of different sizes.
+The following example shows how to declare and initialize a jagged array representing a multiplication table.
+
+```qsharp
+let N = 4;
+mutable multiplicationTable = new Int[][N];
+for (i in 1..N) {
+    mutable row = new Int[i];
+    for (j in 1..i) {
+        set row w/= j-1 <- i * j;
+    }
+    set multiplicationTable w/= i-1 <- row;
```

Comment:
I can't find an explanation of the use of the equal sign for `w/=`... at least in this page, just `w\`

Reply:
Great catch. I think this definitely needs more clarification in the docs in general. It appears that 
- `w/=` is for the "update-and-reassign _statement_", and hence used with mutable variables and the prefacing "set ..."
- `w/` is for the "copy-and-update _expression_", and, among other things, can be used to bind new immutable variables (e.g. a new array that's a copy of another but with a value at an index changed)